java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/DateConsumer.java

```diff
@@ -101,7 +102,7 @@ public NonNullableDateConsumer(DateMilliVector vector, int index, Calendar calen
     public void consume(ResultSet resultSet) throws SQLException {
       Date date = calendar == null ? resultSet.getDate(columnIndexInResultSet) :
```

Comment:
this might be legacy, but `resultSet.getDate` returns `java.sql.Date` instead of `java.util.Date`.
Seems like there might've been some implicit conversion where we don't want underneath?

Reply:
Sure. I agree with you that java.sql.Date is better. 
Here java.sql.Date is converted to java.util.Date because the latter is the super class of the former.